scripts/src/create.ts

```diff
@@ -195,35 +188,121 @@ function getStellarAddresses() {
 	}
 }
 
-initModels(true).then(async () => {
-	const appsDir = process.argv[2];
-	const offersDir = process.argv[3];
+type Config = {
+	apps_dir: string | null;
+	offers_dir: string | null;
+	app_list: string[];
+	no_update: boolean | null;
+	dry_run: boolean | null;
+	require_update_confirm: boolean | null
+	create_db: boolean | null
+};
+
+function initArgsParser(): Config {
+	const ArgumentParser = require("argparse").ArgumentParser;
+	const parser = new ArgumentParser({
+		version: "2.0.0",
+		addHelp: true,
+		description: "DB initializer script for Apps and Offers. Apps configuration come as JSON files and offers as CSV files.",
+		argumentDefault: undefined  // Normalizing defaults, otherwise they're set to {}
+	});
+	parser.addArgument(["--apps-dir"], {
+		help: "Location (directory) of app config json files"
+	});
+	parser.addArgument(["--offers-dir"], {
+		help: "Location (directory) of offers csv files"
+	});
+	parser.addArgument(["--app-list"], {
+		help: "Comma separated list of apps (i.e smpl, swel, kik, test, p365...) to have the earn offers added to (ALL, in caps, to add to all apps)",
+		// choices: ["smpl", "swel", "kik", "test", "p365", "*"]
+	});
+	parser.addArgument(["--no-update"], {
+		help: "Don't update existing earn offers, only create new ones.",
+		action: "storeTrue"
+	});
+	parser.addArgument(["-d", "--dry-run"], {
+		help: "Process the data but don't touch the DB",
+		action: "storeTrue"
+	});
+	parser.addArgument(["-c", "--create-db"], {
+		help: `Create tables/schemes if needed. ${"\x1b[41m" /* red */} USUALLY SHOULD BE NOT RUN IN PRODUCTION${"\x1b[0m" /* reset */}`,
+		action: "storeTrue"
+	});
+
+	// parser.addArgument(["-c", "--require-update-confirm"], {
+	// 	help: "Ask for confirmation before updating earn offers",
+	// 	action: "storeTrue"
+	// });
+	const parsed = parser.parseArgs();
+	parsed.app_list = parsed.app_list ? parsed.app_list.split(" ") : [];
+	return parsed as Config;
+}
+
+function confirmPrompt(message: string) {
+	const readline = require("readline");
+	const prompt = readline.createInterface(process.stdin, process.stdout);
+	return new Promise(resolve => {
+		prompt.question(message + "\n", (answer: string) => {
+			prompt.close();
+			resolve(answer);
+		});
+	});
+}
 
-	for (const filename of fs.readdirSync(path(appsDir))) {
-		if (!filename.endsWith(".json")) {
-			console.info(`skipping non json file ${filename}`);
-			continue;
+config = initArgsParser();
+console.log(config);
```

Comment:
is this a debugging thing or you want to output the config all the time?

Reply:
leftover debugging thing